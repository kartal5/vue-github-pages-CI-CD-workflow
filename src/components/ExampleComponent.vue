<template>
  <div>
    <p>{{ message }}</p>
    <button @click="changeMessage">Change Message</button>
  </div>
</template>

<script>
export default {
  name: 'ExampleComponent',
  data() {
    return {
      message: 'Hello, Vue!'
    }
  },
  methods: {
    changeMessage() {
      this.message = 'Message Changed!'
    }
  }
}
</script>
