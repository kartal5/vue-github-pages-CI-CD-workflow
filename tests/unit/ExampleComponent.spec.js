import { shallowMount } from '@vue/test-utils'
import ExampleComponent from '@/components/ExampleComponent.vue'

describe('ExampleComponent.vue', () => {
  it('renders message when component is created', () => {
    const wrapper = shallowMount(ExampleComponent)
    expect(wrapper.text()).toMatch('Hello, Vue!')
  })

  it('has the correct name', () => {
    const wrapper = shallowMount(ExampleComponent)
    expect(wrapper.vm.$options.name).toMatch('ExampleComponent')
  })

  it('has a message data property', () => {
    const wrapper = shallowMount(ExampleComponent)
    expect(wrapper.vm.message).toBe('Hello, Vue!')
  })

  it('matches the snapshot', () => {
    const wrapper = shallowMount(ExampleComponent)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the correct markup', () => {
    const wrapper = shallowMount(ExampleComponent)
    expect(wrapper.html()).toContain('<p>Hello, Vue!</p>')
  })
})
