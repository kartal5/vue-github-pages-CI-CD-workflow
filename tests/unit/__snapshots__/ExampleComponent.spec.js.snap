// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent.vue matches the snapshot 1`] = `
<div>
  <p>Hello, Vue!</p><button>Change Message</button>
</div>
`;
